server/src/main/java/org/elasticsearch/snapshots/SnapshotShardSizeInfo.java

```diff
@@ -48,6 +48,7 @@ public Long getShardSize(ShardRouting shardRouting) {
 
     public long getShardSize(ShardRouting shardRouting, long fallback) {
```

Comment:
maybe rename this to `getFetchedShardSize`? I am not too fond of having two methods with same name and a fallback argument that behaves differently (thinking of the `!= null` assertion meaning this is only legal to call when size has been fetched).

Reply:
I think the assertion I added is too optimistic, there are no guarantee that the information has been fetched. I removed it.